src/test/java/org/opentripplanner/model/TimetableTest.java

```diff
@@ -591,6 +593,346 @@ public void testUpdateWithRequiredDelayPropagationWhenItsRequired() {
     });
   }
 
+  /**
+   * Test TripUpdate with 1 stop cancellation and the regular stop times after the cancelled stop is
+   * earlier than the scheduled time at the cancelled stop.
+   * Scheduled: 0, 600, 1200
+   * Test case: 0, cancelled, 400
+   * Expect no errors and applied stop times should be increasing.
+   */
+  @Test
+  public void testUpdateWithCancellationAndEarlierThanCancelled() {
+    TripUpdateBuilder tripUpdateBuilder = new TripUpdateBuilder(
+      TRIP_ID,
+      SERVICE_DATE,
+      SCHEDULED,
+      TIME_ZONE
+    )
+      .addDelayedStopTime(1, 0)
+      .addSkippedStop(2)
+      .addDelayedStopTime(3, -800, -800);
+
+    TripUpdate tripUpdate = tripUpdateBuilder.build();
+
+    var patch = timetable.createUpdatedTripTimes(
+      tripUpdate,
+      TIME_ZONE,
+      SERVICE_DATE,
+      BackwardsDelayPropagationType.REQUIRED_NO_DATA
+    );
+
+    assertTrue(patch.isSuccess());
+
+    patch.ifSuccess(p -> {
+      var updatedTripTimes = p.getTripTimes();
+      assertNotNull(updatedTripTimes);
+      assertEquals(0, updatedTripTimes.getArrivalDelay(0));
+      assertEquals(0, updatedTripTimes.getDepartureDelay(0));
+      assertEquals(-800, updatedTripTimes.getArrivalDelay(2));
+      assertEquals(-800, updatedTripTimes.getDepartureDelay(2));
+      assertFalse(updatedTripTimes.isCancelledStop(0));
+      assertTrue(updatedTripTimes.isCancelledStop(1));
+      assertFalse(updatedTripTimes.isCancelledStop(2));
+      assertTrue(updatedTripTimes.getDepartureTime(0) <= updatedTripTimes.getArrivalTime(1));
+      assertTrue(updatedTripTimes.getDepartureTime(1) <= updatedTripTimes.getArrivalTime(2));
+    });
+  }
+
+  /**
+   * Test TripUpdate with 1 stop cancellation and the regular stop times before the cancelled stop is
+   * later than the scheduled time at the cancelled stop.
+   * Scheduled: 0, 600, 1200
+   * Test case: 1000, cancelled, 1200
+   * Expect no errors and applied stop times should be increasing.
+   */
+  @Test
+  public void testUpdateWithCancellationAndLaterThanCancelled() {
+    TripUpdateBuilder tripUpdateBuilder = new TripUpdateBuilder(
+      TRIP_ID,
+      SERVICE_DATE,
+      SCHEDULED,
+      TIME_ZONE
+    )
+      .addDelayedStopTime(1, 1000, 1000)
+      .addSkippedStop(2)
+      .addDelayedStopTime(3, 0, 0);
+
+    TripUpdate tripUpdate = tripUpdateBuilder.build();
+
+    var patch = timetable.createUpdatedTripTimes(
+      tripUpdate,
+      TIME_ZONE,
+      SERVICE_DATE,
+      BackwardsDelayPropagationType.REQUIRED_NO_DATA
+    );
+
+    assertTrue(patch.isSuccess());
+
+    patch.ifSuccess(p -> {
+      var updatedTripTimes = p.getTripTimes();
```

Comment:
Can you please run the `validateNonIncreasingTimes` in the test to make absolutely sure that they are increasing?

Reply:
The `validateNonIncreasingTimes` is called at the end of `createUpdatedTripTimes` method, and the interpolation is done before validating times. If the times are not increasing, the `assertTrue` you referenced here will fail.

```
    // Interpolate missing times from CANCELLED stops. Note: Currently for GTFS-RT ONLY.
    if (interpolateMissingTimes(newTimes)) {
      LOG.debug("Interpolated delays for cancelled stops on trip {}.", tripId);
    }

    // Validate for non-increasing times. Log error if present.
    var error = newTimes.validateNonIncreasingTimes();
```

Reply:
Ah, I see. that's ok then.